qiskit/pulse/instructions/play.py

```diff
@@ -31,31 +33,98 @@ class Play(Instruction):
     cycle time, dt, of the backend.
     """
 
-    def __init__(self, pulse: Pulse, channel: PulseChannel, name: Optional[str] = None):
+    def __init__(
+        self,
+        pulse: Pulse,
+        mixed_frame: Optional[Union[PulseChannel, MixedFrame]] = None,
+        pulse_target: Optional[PulseTarget] = None,
+        frame: Optional[Frame] = None,
+        name: Optional[str] = None,
+        channel: Optional[PulseChannel] = None,
+    ):
         """Create a new pulse instruction.
 
         Args:
             pulse: A pulse waveform description, such as
                    :py:class:`~qiskit.pulse.library.Waveform`.
-            channel: The channel to which the pulse is applied.
+            mixed_frame: A ``MixedFrame`` on which the pulse is to be applied.
+            pulse_target: A ``PulseTarget`` on which the pulse is to be applied. Has to be
+                provided when frame is provided, and can not be provided if ``MixedFrame``
+                was provided.
+            frame: The ``Frame`` of the carrier wave. Has to be provided when pulse_target is
+                provided, and can not be provided if ``MixedFrame`` was provided.
+            channel: The channel to which the pulse is applied. Can not be provided
+                if ``MixedFrame`` or ``Frame`` and ``PulseTarget`` were provided.
             name: Name of the instruction for display purposes. Defaults to ``pulse.name``.
+
+        Raises:
+            PulseError: If ``frame`` and ``pulse_target`` are not provided together.
+            PulseError: If ``mixed_frame`` is provided together with ``frame`` and ``pulse_target``.
+            PulseError: If ``channel`` is provided together with ``mixed_frame``
+                or ``frame`` and ``pulse_target``.
         """
+        if isinstance(mixed_frame, PulseChannel):
```

Comment:
This can be replaced with `deprecate_arg` decorator with `predicate` pattern.

Reply:
I wanted to test for both `channel` and `name` in the same place to avoid confusion. I don't think you can do that with `deprecate_arg`.